### PR TITLE
[systemtest] Load ST variables from json config

### PR DESCRIPTION
**Pull request by kornys**
Signed-off-by: David Kornel <kornys@outlook.com>

### Type of change

- Enhancement

### Description

Load variables in ST from json config

example of config

```
{
  "DOCKER_TAG": "dev",
  "TEST_LOG_DIR": "/tmp/testlogs",
}
```

### Checklist

- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -181,6 +181,14 @@ All available test groups are listed in [Constants](systemtest/src/main/java/io/
 ## Environment variables
 
 System tests can be configured by several environment variables, which are loaded before test execution.
+Variables can be defined via env or via configuration file, this file can be placed anywhere and path to this
+file is defined by env variable `CONFIG`, if `CONFIG` env variable is not defined, default config location is used `systemtests/config.json`.
```
Comment (Frawless) [inline development-docs/TESTING.md:185]:
```suggestion
file is defined by env variable `CONFIG`, if `CONFIG` env variable is not defined, default config location is used `systemtest/config.json`.
```

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only

**Comment by scholzj** [comment]
@Frawless @kornys Are the two failed tests something to be worried about? Or just some flaky tests?

**Comment by Frawless** [comment]
I will check what's wrong here. It could be an env issue, but I am not sure yet.

**Comment by kornys** [comment]
@strimzi-ci run tests profile=acceptance testcase=io.strimzi.systemtest.operators.user.UserST#testUpdateUser,io.strimzi.systemtest.tracing.TracingST#testProducerConsumerStreamsService,io.strimzi.systemtest.connect.ConnectS2IST#testKafkaConnectorWithConnectS2IAndConnectWithSameName test_only

File: development-docs/TESTING.md [hunk 2]
```diff
@@ -181,6 +181,14 @@ All available test groups are listed in [Constants](systemtest/src/main/java/io/
 ## Environment variables
 
 System tests can be configured by several environment variables, which are loaded before test execution.
+Variables can be defined via env or via configuration file, this file can be placed anywhere and path to this
+file is defined by env variable `CONFIG`, if `CONFIG` env variable is not defined, default config location is used `systemtest/config.json`.
+
+Loading of variables has following priority:
+1. Environment variable
+2. Variable defined in config
+3. Default value
```
Comment (samuel-hawker) [inline development-docs/TESTING.md:190]:
```suggestion
Variables can be defined via environmental variables or a configuration file, this file can be located anywhere on the file system as long as a path is provided to this file. 
The path is defined by environmental variable `SYSTEM_TEST_CONFIG_PATH`, if the `SYSTEM_TEST_CONFIG_PATH` environmental variable is not defined, the default config file location is used `systemtest/config.json`.

Loading of system configuration has the following priority order:
1. Environment variable
2. Variable defined in configuration file
3. Default value
```
I recommend CONFIG -> `SYSTEM_TEST_CONFIG_PATH` as it's less likely to conflict with pre-set envars on running systems.

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java [hunk 3]
```diff
@@ -4,18 +4,33 @@
...
 public class Environment {
 
     private static final Logger LOGGER = LogManager.getLogger(Environment.class);
+    private static final Map<String, String> VALUES = new HashMap<>();
+    private static final JsonNode JSON_DATA = loadJsonEnv();
 
+    /**
+     * Specify config json/yaml path with env variables values
+     */
+    private static final String CONFIG_ENV = "CONFIG";
```
Comment (samuel-hawker) [inline systemtest/src/main/java/io/strimzi/systemtest/Environment.java:33]:
```suggestion
    private static final String CONFIG_FILE_PATH_ENVAR = "SYSTEM_TEST_CONFIG_PATH";
```

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java [hunk 4]
```diff
@@ -4,18 +4,33 @@
...
  * Class which holds environment variables for system tests.
  */
 public class Environment {
 
     private static final Logger LOGGER = LogManager.getLogger(Environment.class);
+    private static final Map<String, String> VALUES = new HashMap<>();
+    private static final JsonNode JSON_DATA = loadJsonEnv();
 
+    /**
+     * Specify config json/yaml path with env variables values
```
Comment (samuel-hawker) [inline systemtest/src/main/java/io/strimzi/systemtest/Environment.java:31]:
```suggestion
     * Specify the system test configuration file path from an environmental variable
     * This configuration file has a JSON or YAML format
```

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java [hunk 5]
```diff
@@ -161,4 +164,35 @@ public static boolean isHelmInstall() {
...
+                        null);
+        T returnValue = defaultValue;
+        if (value != null) {
+            returnValue = converter.apply(value);
+        }
+        VALUES.put(var, String.valueOf(returnValue));
+        return returnValue;
+    }
+
+    private static JsonNode loadJsonEnv() {
```
Comment (samuel-hawker) [inline systemtest/src/main/java/io/strimzi/systemtest/Environment.java:186]:
```suggestion
    private static JsonNode loadConfigurationFile() {
```
I would recommend this renaming to make clearer what it is doing, the fact that the file is JSON/YAML is an implementation detail and (hopefully) should be obvious from the documentation + return type

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only

**Comment by Frawless** [comment]
@scholzj Failures were caused by environment issues from my POV, I wasn't able to reproduce it.

**Comment by scholzj** [comment]
Thanks for the PR @kornys

**Pull request merged**